protocol_connect.go

```diff
@@ -74,13 +74,9 @@ func (*protocolConnect) NewHandler(params *protocolHandlerParams) protocolHandle
 
 // NewClient implements protocol, so it must return an interface.
 func (*protocolConnect) NewClient(params *protocolClientParams) (protocolClient, error) {
-	url, err := validateRequestURL(params.URL)
```

Comment:
We do this validate/parse step now when creating the actual Client before we even get here, so there's no reason to do any validation per-protocol.